modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -16,33 +16,42 @@ def create_a_record(fqdn, ip)
       if found = dns_find(fqdn)
         raise(Proxy::Dns::Collision, "#{fqdn} is already used by #{found}") if found != ip
       else
-        zone = match_zone(fqdn)
+        zone = match_zone(fqdn, enum_zones)
         msg = "Added DNS entry #{fqdn} => #{ip}"
         cmd = "/RecordAdd #{zone} #{fqdn}. A #{ip}"
         execute(cmd, msg)
+        nil
       end
     end
 
-    # noop
-    def create_ptr_record(fqdn, ip)
-      found = dns_find(ip)
-      raise(Proxy::Dns::Collision, "#{ip} is already used by #{found}") if found && found != fqdn
-      true
+    def create_ptr_record(fqdn, ptr)
+      ip = ptr_to_ip ptr
+      if found = dns_find(ip)
```

Comment:
Sigh. This is broken, because ```dns_find``` reverses ip address before calling ```Resolv::DNS.getname```, which then returns false and creation of a ptr record fails, event though it should've succeeded.

I would suggest:
 - copy dns_find code into this module
 - break it up into find_ptr and find_fqdn
 - fix issues with multiple mappings

We'll move the changes into dns_common once the dust settles here.